course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala

```diff
@@ -189,6 +194,31 @@ class ContentConsumptionActor @Inject() extends BaseEnrolmentActor {
         response.getResult.getOrDefault(JsonKey.RESPONSE, new java.util.ArrayList[java.util.Map[String, AnyRef]]).asInstanceOf[java.util.List[java.util.Map[String, AnyRef]]]
     }
 
+    //Added content progress
+    def getContentsProgress(userId: String, courseId : String, contentIds: java.util.List[String], batchId: String, requestContext: RequestContext):java.util.List[java.util.Map[String, AnyRef]] = {
+        val filters = new java.util.HashMap[String, AnyRef]() {{
+            put("userid", userId)
+            put("courseid", courseId)
+            put("batchid", batchId)
+            if(CollectionUtils.isNotEmpty(contentIds))
+                put("contentid", contentIds)
+        }}
+        val response = cassandraOperation.getBlobAsText(requestContext, consumptionDBInfo.getKeySpace, "user_content_consumption_progress", filters, util.Arrays.asList("body") )
```

Comment:
@pritha-tarento : why didn't we use "map" data type for progressDetails instead of a "blob"?

Reply:
map data type would be able to store value as text form. Hence to address blob was suggested.